blocks/api/serializer.js

```diff
@@ -15,15 +15,45 @@ import { Component, createElement, renderToString, cloneElement, Children } from
 import { applyFilters } from '../hooks';
 import { getBlockType, getUnknownTypeHandlerName } from './registration';
 
+/**
+ * Returns the block's name with common prefixes: 'core/' or 'core-' (in 'core-embed/') dropped
+ *
+ * @param {String}   blockName  The block name
+ * @return {string}             Friendly name
+ */
+function friendlyBlockName( blockName ) {
+	return blockName.replace( /\//, '-' ).replace( /^core-/, '' );
+}
+
 /**
  * Returns the block's default classname from its name
  *
  * @param {String}   blockName  The block name
  * @return {string}             The block's default class
  */
 export function getBlockDefaultClassname( blockName ) {
-	// Drop common prefixes: 'core/' or 'core-' (in 'core-embed/')
-	return 'wp-block-' + blockName.replace( /\//, '-' ).replace( /^core-/, '' );
+	return `wp-block-${ friendlyBlockName( blockName ) }`;
```

Comment:
Not sure it's worth concerning about, but there's a decent amount of redundancy between the class name and style applied to a block with custom styles, i.e.:

```html
<p id="wp-block-paragraph-z60dssg" class="wp-block-paragraph">
</p>
```

Reply:
I agree the naming was redundant, I changed the id's so now they are in format 'paragraph-....'.